src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -1833,6 +1834,20 @@ public IndexDefinition GetIndexDefinition()
 
                             var retriever = GetQueryResultRetriever(query, documentsContext, fieldsToFetch, includeDocumentsCommand);
 
+                            if (query.Metadata.WhereMethods.TryGetValue(QueryMetadata.WhereMethod.RightSideMethod, out var method))
+                            {
+                                if (query.Metadata.Query.Where is BinaryExpression be && be.Right is MethodExpression me)
+                                {
+                                    var value = method.EvaluateSingleMethod((QueryResultRetrieverBase)retriever, null);
+                                    query.QueryParameters.Modifications = new DynamicJsonValue
+                                    {
+                                        [me.Name] = value
+                                    };
+                                    query.QueryParameters = documentsContext.ReadObject(query.QueryParameters, "add-parameter");
+                                    be.Right = new ValueExpression(me.Name, ValueTokenType.Parameter);
```

Comment:
You are changing QueryMetadata here. Changing `be.Right` to `ValueExpression` will mean that you will never go into this if-statement again for future queries.